src/ui-test/utils/waitConditions.ts

```diff
@@ -104,3 +95,83 @@ export async function updateFileText(oldText: string, newText: string, timePerio
         return false;
     }
 }
+
+export async function webViewHasTextInWebElement(driver: WebDriver, text: string, locator: Locator = { id: 'content' }, timePeriod = 1000, timeout = 25000): Promise<boolean> {
+    let webView = new WebView();
+    try {
+        return await driver.wait(async () => {
+            try {
+                webView = new WebView();
+                if (await webView.isDisplayed() && await webView.isEnabled()) {
+                    await webView.switchToFrame();
+                    const contentElement = await webView.findWebElement(locator);
+                    const content = await contentElement.getText();
+                    if (content.indexOf(text) > -1) {
+                        await webView.switchBack();
+                        return true;
+                    } else {
+                        await webView.switchBack();
+                        await webView.getDriver().sleep(timePeriod);
```

Comment:
why do we need to provide a sleep here? Shouldn't providing another parameter to configure the wait interval enough?

Reply:
I actually test updates without sleep - https://app.circleci.com/pipelines/github/unsortedhashsets/vscode-camelk/57/workflows/7845102c-b55e-4fc6-b2fc-3373fc3cda21/jobs/73/parallel-runs/0/steps/0-111

Hope we can postpone that update to the next PR (cause it is in stabilization actually), I am a bit afraid for the stability of the test suite

Reply:
Due fix of the flaky ui-test I have decided to integrate here already implemented features from next PR, so all static sleep/wait were deleted

Done

Reply:
Except for text elements:

- `webViewHasTextInWebElement`
- `outputViewHasText`

The timer in the default example is set on 1s, so it is not necessary to check output more often